Index: lib/Styler/Util.js
===================================================================
--- lib/Styler/Util.js	(revision 238)
+++ lib/Styler/Util.js	(revision 284)
@@ -19,7 +19,7 @@
      *     three)
      */
     getSymbolTypeFromRule: function(rule){
-        var symbolizer = rule.symbolizer
+        var symbolizer = rule.symbolizer;
         if (symbolizer["Line"] || symbolizer["Point"] || symbolizer["Polygon"]) {
             for (var type in symbolizer) {
                 if (type != "Text") {
@@ -28,6 +28,4 @@
             }
         }
     }
-}
-
-
+};
