diff --git src/main/webapp/lib/styler/lib/Styler/widgets/FilterBuilder.js src/main/webapp/lib/styler/lib/Styler/widgets/FilterBuilder.js
index a2ee51d..4c3f969 100644
--- src/main/webapp/lib/styler/lib/Styler/widgets/FilterBuilder.js
+++ src/main/webapp/lib/styler/lib/Styler/widgets/FilterBuilder.js
@@ -495,6 +495,26 @@ Styler.FilterBuilder = Ext.extend(Ext.Panel, {
                         }
                     }
                     break;
+                case OpenLayers.Filter.Comparison.BETWEEN:
+                    console.log(filter.lowerBoundary);
+                    console.log(filter.upperBoundary);
+                    filter = new OpenLayers.Filter.Logical({
+                        type: OpenLayers.Filter.Logical.AND,
+                        filters: [
+                            new OpenLayers.Filter.Comparison({
+                                type: OpenLayers.Filter.Comparison.GREATER_THAN_OR_EQUAL_TO,
+                                property: filter.property,
+                                value: filter.lowerBoundary
+                            }),
+                            new OpenLayers.Filter.Comparison({
+                                type: OpenLayers.Filter.Comparison.LESS_THAN_OR_EQUAL_TO,
+                                property: filter.property,
+                                value: filter.upperBoundary
+                            })
+                        ]
+                    });
+                    filter = this.customizeFilter(filter);
+                    break;
                 default:
                     // non-logical filters get wrapped
                     filter = this.wrapFilter(filter);
