Index: restconfig/src/main/java/org/geoserver/catalog/rest/CoverageStoreFileResource.java
===================================================================
--- restconfig/src/main/java/org/geoserver/catalog/rest/CoverageStoreFileResource.java	(revision 15322)
+++ restconfig/src/main/java/org/geoserver/catalog/rest/CoverageStoreFileResource.java	(working copy)
@@ -136,6 +136,7 @@
             }
             
             CoverageInfo cinfo = builder.buildCoverage( reader );
+            cinfo.setTitle(cinfo.getName());
             
             //check if the name of the coverage was specified
             String coverageName = form.getFirstValue("coverageName");
Index: restconfig/src/main/java/org/geoserver/catalog/rest/DataStoreFileResource.java
===================================================================
--- restconfig/src/main/java/org/geoserver/catalog/rest/DataStoreFileResource.java	(revision 15322)
+++ restconfig/src/main/java/org/geoserver/catalog/rest/DataStoreFileResource.java	(working copy)
@@ -89,8 +89,13 @@
         };
         getResponse().setEntity( fmt.toRepresentation( directory ) );
     }
-    
+
     @Override
+    protected File findPrimaryFile(File directory, String format) {
+        return super.findPrimaryFile(directory, format);    //To change body of overridden methods use File | Settings | File Templates.
+    }
+
+    @Override
     public void handlePut() {
         String workspace = (String)getRequest().getAttributes().get("workspace");
         String datastore = (String)getRequest().getAttributes().get("datastore");
@@ -265,6 +270,7 @@
                 if ( add ) {
                     //auto configure the feature type as well
                     ftinfo = builder.buildFeatureType(fs);
+                    ftinfo.setTitle(fs.getSchema().getName().getLocalPart());
                 }
                 else {
                     ftinfo = catalog.getFeatureTypeByName( namespace.getPrefix(), featureTypeNames[i] );
Index: restconfig/src/main/java/org/geoserver/catalog/rest/StoreFileFinder.java
===================================================================
--- restconfig/src/main/java/org/geoserver/catalog/rest/StoreFileFinder.java	(revision 15322)
+++ restconfig/src/main/java/org/geoserver/catalog/rest/StoreFileFinder.java	(working copy)
@@ -21,6 +21,7 @@
     protected static HashMap<String,String> formatToDataStoreFactory = new HashMap();
     static {
         formatToDataStoreFactory.put( "shp", "org.geotools.data.shapefile.ShapefileDataStoreFactory");
+        formatToDataStoreFactory.put( "mif", "org.geotools.data.mif.MIFDataStoreFactory");
         formatToDataStoreFactory.put( "properties", "org.geotools.data.property.PropertyDataStoreFactory");
     }
     
