Index: lib/Styler/widgets/FilterBuilder.js
===================================================================
--- lib/Styler/widgets/FilterBuilder.js	(revision 284)
+++ lib/Styler/widgets/FilterBuilder.js	(working copy)
@@ -107,7 +107,7 @@
      */
     createToolBar: function() {
         var bar = [{
-            text: "add condition",
+            text: "ajouter une condition",
             iconCls: "add",
             handler: function() {
                 this.addCondition();
@@ -116,7 +116,7 @@
         }];
         if(this.allowGroups) {
             bar.push({
-                text: "add group",
+                text: "ajouter un groupe",
                 iconCls: "add",
                 handler: function() {
                     this.addCondition(true);
@@ -533,4 +533,4 @@
 Styler.FilterBuilder.NONE_OF = 2;
 Styler.FilterBuilder.NOT_ALL_OF = 3;
 
-Ext.reg('gx_filterbuilder', Styler.FilterBuilder); 
\ No newline at end of file
+Ext.reg('gx_filterbuilder', Styler.FilterBuilder); 
