Index: lib/Styler/widgets/form/ComparisonComboBox.js
===================================================================
--- lib/Styler/widgets/form/ComparisonComboBox.js	(révision 344)
+++ lib/Styler/widgets/form/ComparisonComboBox.js	(copie de travail)
@@ -3,7 +3,7 @@
  */
 
 Ext.namespace("Styler.form");
-Styler.form.ComarisonComboBox = Ext.extend(Ext.form.ComboBox, {
+Styler.form.ComparisonComboBox = Ext.extend(Ext.form.ComboBox, {
     
     allowedTypes: [
         [OpenLayers.Filter.Comparison.EQUAL_TO, "="],
@@ -37,8 +37,8 @@
         };
         Ext.applyIf(this, defConfig);
         
-        Styler.form.ComarisonComboBox.superclass.initComponent.call(this);
+        Styler.form.ComparisonComboBox.superclass.initComponent.call(this);
     }
 });
 
-Ext.reg("gx_comparisoncombo", Styler.form.ComarisonComboBox);
\ No newline at end of file
+Ext.reg("gx_comparisoncombo", Styler.form.ComparisonComboBox);
\ No newline at end of file
