Index: lib/Styler/widgets/FilterBuilder.js
===================================================================
--- lib/Styler/widgets/FilterBuilder.js	(révision 346)
+++ lib/Styler/widgets/FilterBuilder.js	(copie de travail)
@@ -109,6 +109,7 @@
         this.bbar = this.createToolBar();
         
         if (this.allowSpatial) {
+            this.controls = [];
             var layer = this.createVectorLayer();
             this.tbar = this.createEditionToolBar(layer);
             this.vectorLayer = layer;
@@ -142,6 +143,13 @@
         Styler.FilterBuilder.superclass.initComponent.call(this);
     },
     
+    deactivateControls: function() {
+        var controls = this.controls;
+        for (var i=0,l=controls.length; i<l; i++) {
+            controls[i].deactivate();
+        }
+    },
+    
     setUp: function() {
         if (this.vectorLayer) {
             this.vectorLayer.setVisibility(true);
@@ -149,6 +157,7 @@
     },
     
     tearDown: function() {
+        this.deactivateControls();
         if (this.vectorLayer) {
             this.vectorLayer.setVisibility(false);
         }
@@ -194,27 +203,29 @@
      * Method: createEditionToolBar
      */
     createEditionToolBar: function(layer) {
-        var createDrawControl = function(handler) {
-            return new OpenLayers.Control.DrawFeature(layer, handler);
+        var createDrawControl = function(handler, controls) {
+            var control = new OpenLayers.Control.DrawFeature(layer, handler)
+            controls.push(control);
+            return control;
         };
         return new Ext.Toolbar({
             items: [
                 new GeoExt.Action({
-                    control: createDrawControl(OpenLayers.Handler.Point),
+                    control: createDrawControl(OpenLayers.Handler.Point, this.controls),
                     map: this.map,
                     toggleGroup: "querier",
                     tooltip: "Ajouter une condition géographique sur la base d'un point",
                     iconCls: "drawpoint"
                 }),
                 new GeoExt.Action({
-                    control: createDrawControl(OpenLayers.Handler.Path),
+                    control: createDrawControl(OpenLayers.Handler.Path, this.controls),
                     map: this.map,
                     toggleGroup: "querier",
                     tooltip: "Ajouter une condition géographique sur la base d'une ligne",
                     iconCls: "drawline"
                 }),
                 new GeoExt.Action({
-                    control: createDrawControl(OpenLayers.Handler.Polygon),
+                    control: createDrawControl(OpenLayers.Handler.Polygon, this.controls),
                     map: this.map,
                     toggleGroup: "querier",
                     tooltip: "Ajouter une condition géographique sur la base d'un polygone",
