Index: src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js
===================================================================
--- src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js	(revision 365)
+++ src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js	(working copy)
@@ -4,8 +4,6 @@
 
 /**
  * @include Styler/widgets/tips/SliderTip.js
- * @include Styler/widgets/form/ColorField.js
- * @include Styler/ColorManager.js
  */
 
 Ext.namespace("Styler");
@@ -43,11 +41,10 @@
                 width: 100 // TODO: move to css
             },
             items: [{
-                xtype: "gx_colorfield",
+                xtype: "ext.ux.colorpicker",
                 fieldLabel: "Color",
                 name: "color",
                 value: this.symbolizer["fillColor"],
-                plugins: new Styler.ColorManager(),
                 listeners: {
                     valid: function(field) {
                         this.symbolizer["fillColor"] = field.getValue();
@@ -97,4 +94,4 @@
     
 });
 
-Ext.reg('gx_fillsymbolizer', Styler.FillSymbolizer); 
\ No newline at end of file
+Ext.reg('gx_fillsymbolizer', Styler.FillSymbolizer); 
Index: src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
===================================================================
--- src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js	(revision 365)
+++ src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js	(working copy)
@@ -4,8 +4,6 @@
 
 /**
  * @include Styler/widgets/tips/SliderTip.js
- * @include Styler/widgets/form/ColorField.js
- * @include Styler/ColorManager.js
  */
 
 Ext.namespace("Styler");
@@ -74,11 +72,10 @@
                     scope: this
                 }
             }, {
-                xtype: "gx_colorfield",
+                xtype: "ext.ux.colorpicker",
                 name: "color",
                 fieldLabel: "Color",
                 value: this.symbolizer["strokeColor"],
-                plugins: new Styler.ColorManager(),
                 listeners: {
                     valid: function(field) {
                         this.symbolizer["strokeColor"] = field.getValue();
@@ -140,4 +137,4 @@
     
 });
 
-Ext.reg('gx_strokesymbolizer', Styler.StrokeSymbolizer); 
\ No newline at end of file
+Ext.reg('gx_strokesymbolizer', Styler.StrokeSymbolizer); 
