Index: lib/Styler/widgets/RulePanel.js
===================================================================
--- lib/Styler/widgets/RulePanel.js	(revision 412)
+++ lib/Styler/widgets/RulePanel.js	(working copy)
@@ -136,24 +136,28 @@
             }
         });
         
-        var maxScaleDenominator = this.rule.maxScaleDenominator != null ?
-                                  this.rule.maxScaleDenominator : undefined;
-        var maxScaleLimit = this.maxScaleLimit != null ?
-                            this.maxScaleLimit : undefined;
-        var limitMaxScale = !!maxScaleDenominator || !!maxScaleLimit;
-        var minScaleDenominator = this.rule.minScaleDenominator != null ?
-                                  this.rule.minScaleDenominator : undefined;
-        var minScaleLimit = this.minScaleLimit != null ?
-                            this.minScaleLimit : undefined;
-        var limitMaxScale = !!minScaleDenominator || !!minScaleLimit;
+        var maxScaleDenominator, maxScaleLimit;
+        var minScaleDenominator, minScaleLimit;
+        maxScaleDenominator =
+            this.rule.maxScaleDenominator != null ?
+                this.rule.maxScaleDenominator : undefined;
+        maxScaleLimit =
+            this.maxScaleLimit != null ?
+                this.maxScaleLimit : undefined;
+        minScaleDenominator =
+            this.rule.minScaleDenominator != null ?
+                this.rule.minScaleDenominator : undefined;
+        minScaleLimit =
+            this.minScaleLimit != null ?
+                this.minScaleLimit : undefined;
 
         this.scaleLimitPanel = new Styler.ScaleLimitPanel({
             maxScaleDenominator: maxScaleDenominator,
             maxScaleLimit: maxScaleLimit,
-            limitMaxScale: limitMaxScale,
+            limitMaxScale: !!maxScaleDenominator,
             minScaleDenominator: minScaleDenominator,
             minScaleLimit: minScaleLimit,
-            limitMinScale: limitMaxScale,
+            limitMinScale: !!minScaleDenominator,
             scaleLevels: this.scaleLevels,
             scaleSliderTemplate: this.scaleSliderTemplate,
             modifyScaleTipContext: this.modifyScaleTipContext,
