Index: lib/Styler/widgets/FilterPanel.js
===================================================================
--- lib/Styler/widgets/FilterPanel.js	(révision 427)
+++ lib/Styler/widgets/FilterPanel.js	(copie de travail)
@@ -36,6 +36,7 @@
             triggerAction: "all",
             hideLabel: true,
             allowBlank: false,
+            blankText: "Ce champ est nécessaire",
             displayField: "name",
             valueField: "name",
             value: this.filter.property,
@@ -82,6 +83,7 @@
                 items: [{
                     xtype: "gx_comparisoncombo",
                     value: this.filter.type,
+                    blankText: "Ce champ est nécessaire",
                     listeners: {
                         select: function(combo, record) {
                             this.filter.type = record.get("value");
@@ -96,6 +98,7 @@
                     width: 120,
                     value: this.filter.value,
                     allowBlank: false,
+                    blankText: "Ce champ est nécessaire",
                     listeners: {
                         change: function(el, value) {
                             this.filter.value = value;
Index: lib/Styler/widgets/SpatialFilterPanel.js
===================================================================
--- lib/Styler/widgets/SpatialFilterPanel.js	(révision 427)
+++ lib/Styler/widgets/SpatialFilterPanel.js	(copie de travail)
@@ -43,6 +43,7 @@
         var defComboConfig = {
             xtype: "gx_spatialcombo",
             value: this.filter.type,
+            blankText: "Ce champ est nécessaire",
             listeners: {
                 select: function(combo, record) {
                     this.filter.type = record.get("value");
@@ -109,6 +110,7 @@
                 items: [{
                     xtype: "button", 
                     iconCls: cls,
+                    tooltip: "Modifier la géométrie",
                     enableToggle: true,
                     toggleGroup: this.toggleGroup,
                     listeners: {
Index: lib/Styler/widgets/FilterBuilder.js
===================================================================
--- lib/Styler/widgets/FilterBuilder.js	(révision 427)
+++ lib/Styler/widgets/FilterBuilder.js	(copie de travail)
@@ -631,7 +631,7 @@
                 columnWidth: 0.1,
                 items: [{
                     xtype: "button",
-                    tooltip: "remove condition",
+                    tooltip: "Supprimer cette condition",
                     cls: 'x-btn-icon',
                     iconCls: "delete",
                     handler: function() {
