diff --git lib/Styler/widgets/ScaleLimitPanel.js lib/Styler/widgets/ScaleLimitPanel.js
index 06ee542..235d4aa 100644
--- lib/Styler/widgets/ScaleLimitPanel.js
+++ lib/Styler/widgets/ScaleLimitPanel.js
@@ -163,6 +163,7 @@ Styler.ScaleLimitPanel = Ext.extend(Ext.Panel, {
             fieldLabel: "1",
             value: Math.round(this.maxScaleDenominator),
             disabled: !this.limitMaxScale,
+            validationDelay: 1500, // 1.5 sec
             listeners: {
                 valid: function(field) {
                     var value = Number(field.getValue());
@@ -185,6 +186,7 @@ Styler.ScaleLimitPanel = Ext.extend(Ext.Panel, {
             fieldLabel: "1",
             value: Math.round(this.minScaleDenominator),
             disabled: !this.limitMinScale,
+            validationDelay: 1500, // 1.5 sec
             listeners: {
                 valid: function(field) {
                     var value = Number(field.getValue());
