diff --git src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
index e05b227..9865e0c 100644
--- src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
+++ src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
@@ -31,7 +31,7 @@ Styler.StrokeSymbolizer = Ext.extend(Ext.FormPanel, {
      *     display name.  Default is [["solid", "solid"], ["dash", "dash"],
      *     ["dot", "dot"]].
      */
-    dashStyles: [["solid", "plein"], ["dash", "tiret"], ["dot", "point"]],
+    dashStyles: [["solid", "plein"], ["4 4", "tiret"], ["2 4", "point"]],
     
     border: false,
     
