diff --git src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js
index ceabcbb..749a657 100644
--- src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js
+++ src/main/webapp/lib/styler/lib/Styler/widgets/FillSymbolizer.js
@@ -72,7 +72,18 @@ Styler.FillSymbolizer = Ext.extend(Ext.FormPanel, {
                         }
                     })
                 ]
-            }]
+            }],
+            listeners: {
+                "collapse": function() {
+                    this.symbolizer["fill"] = false;
+                    this.fireEvent("change", this.symbolizer);
+                },
+                "expand": function() {
+                    this.symbolizer["fill"] = true;
+                    this.fireEvent("change", this.symbolizer);
+                },
+                scope: this
+            }
         }];
 
         this.addEvents(
diff --git src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
index 9865e0c..aeb9260 100644
--- src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
+++ src/main/webapp/lib/styler/lib/Styler/widgets/StrokeSymbolizer.js
@@ -115,7 +115,18 @@ Styler.StrokeSymbolizer = Ext.extend(Ext.FormPanel, {
                         }
                     })
                 ]
-            }]
+            }],
+            listeners: {
+                "collapse": function() {
+                    this.symbolizer["stroke"] = false;
+                    this.fireEvent("change", this.symbolizer);
+                },
+                "expand": function() {
+                    this.symbolizer["stroke"] = true;
+                    this.fireEvent("change", this.symbolizer);
+                },
+                scope: this
+            }
         }];
 
         this.addEvents(
