Index: lib/Styler/widgets/FilterBuilder.js
===================================================================
--- lib/Styler/widgets/FilterBuilder.js	(révision 1196)
+++ lib/Styler/widgets/FilterBuilder.js	(copie de travail)
@@ -249,13 +249,22 @@
     createEditingMenu: function(layer) {
         var createDrawControl = function(handler, controls) {
             var control = new OpenLayers.Control.DrawFeature(layer, handler);
+            control.events.on({
+                "activate": function() {
+                    OpenLayers.Element.addClass(this.map.viewPortDiv, "olDrawBox");
+                },
+                "deactivate": function() {
+                    OpenLayers.Element.removeClass(this.map.viewPortDiv, "olDrawBox");
+                },
+                scope: this
+            });
             controls.push(control);
             return control;
         };
         var items = [];
         if (this.geometryTypes.indexOf('point')>-1) {
             items.push(new Ext.menu.CheckItem(new GeoExt.Action({
-                control: createDrawControl(OpenLayers.Handler.Point, this.controls),
+                control: createDrawControl.call(this, OpenLayers.Handler.Point, this.controls),
                 map: this.map,
                 group: "querier",
                 text: "sur la base d'un point",
@@ -264,7 +273,7 @@
         }
         if (this.geometryTypes.indexOf('line')>-1) {
             items.push(new Ext.menu.CheckItem(new GeoExt.Action({
-                control: createDrawControl(OpenLayers.Handler.Path, this.controls),
+                control: createDrawControl.call(this, OpenLayers.Handler.Path, this.controls),
                 map: this.map,
                 group: "querier",
                 text: "sur la base d'une ligne",
@@ -273,7 +282,7 @@
         }
         if (this.geometryTypes.indexOf('polygon')>-1) {
             items.push(new Ext.menu.CheckItem(new GeoExt.Action({
-                control: createDrawControl(OpenLayers.Handler.Polygon, this.controls),
+                control: createDrawControl.call(this, OpenLayers.Handler.Polygon, this.controls),
                 map: this.map,
                 group: "querier",
                 text: "sur la base d'un polygone",
