Index: lib/Styler/widgets/FilterPanel.js
===================================================================
--- lib/Styler/widgets/FilterPanel.js	(révision 1209)
+++ lib/Styler/widgets/FilterPanel.js	(copie de travail)
@@ -300,6 +300,7 @@
         return [{
             layout: "column",
             border: false,
+            height: 25,
             defaults: {border: false},
             items: [{
                 items: [this.attributesComboConfig],
Index: lib/Styler/widgets/FilterBuilder.js
===================================================================
--- lib/Styler/widgets/FilterBuilder.js	(révision 1209)
+++ lib/Styler/widgets/FilterBuilder.js	(copie de travail)
@@ -120,7 +120,7 @@
         this.items = [
             {
                 xtype: "panel",
-                height: 35,
+                height: 30,
                 layout: 'fit',
                 border: false,
                 items: [{
@@ -136,6 +136,8 @@
                         cls: "x-form-item",
                         style: "padding: 0.3em 0.5em 0;" // TODO: replace with css
                     }, {
+                        height: 25,
+                        width: 85,
                         items: [this.createBuilderTypeCombo()]
                     }, {
                         html: this.postComboText,
@@ -713,7 +715,7 @@
                 },
                 scope: this
             },
-            width: 85 // TODO: move to css
+            width: 65 
         };
     },
     
