Index: lib/Styler/widgets/StrokeSymbolizer.js
===================================================================
--- lib/Styler/widgets/StrokeSymbolizer.js	(révision 1255)
+++ lib/Styler/widgets/StrokeSymbolizer.js	(copie de travail)
@@ -31,7 +31,7 @@
      *     display name.  Default is [["solid", "solid"], ["dash", "dash"],
      *     ["dot", "dot"]].
      */
-    dashStyles: [["solid", "plein"], ["4 4", "tiret"], ["2 4", "point"]],
+    dashStyles: [["plein", "plein"], ["4 4", "tiret"], ["2 4", "point"]],
     
     border: false,
     
@@ -59,14 +59,16 @@
                 }),
                 displayField: "display",
                 valueField: "value",
-                value: this.symbolizer["strokeDashstyle"],
+                value: this.symbolizer["strokeDashstyle"] ? 
+                    this.symbolizer["strokeDashstyle"] : "plein",
                 mode: "local",
                 allowBlank: true,
                 triggerAction: "all",
                 editable: false,
                 listeners: {
                     select: function(combo, record) {
-                        this.symbolizer["strokeDashstyle"] = record.get("value");
+                        var v = (record.get("value") == "plein") ? undefined : record.get("value");
+                        this.symbolizer["strokeDashstyle"] = v;
                         this.fireEvent("change", this.symbolizer);
                     },
                     scope: this
