Index: src/wcs1_0/src/main/java/org/geoserver/wcs/DefaultWebCoverageService100.java
===================================================================
--- src/wcs1_0/src/main/java/org/geoserver/wcs/DefaultWebCoverageService100.java	(revision 16592)
+++ src/wcs1_0/src/main/java/org/geoserver/wcs/DefaultWebCoverageService100.java	(working copy)
@@ -519,7 +519,7 @@
             GeneralEnvelope destinationEnvelope = (GeneralEnvelope) getHorizontalEnvelope(computeIntersectionEnvelope(
                     requestedEnvelope, nativeEnvelope));
             if(targetCRS != null) {
-                MathTransform mt = CRS.findMathTransform(nativeCRS, targetCRS);
+                MathTransform mt = CRS.findMathTransform(nativeCRS, targetCRS, true);
                 destinationEnvelope = CRS.transform(mt, destinationEnvelope);
                 destinationEnvelope.setCoordinateReferenceSystem(targetCRS);
             }
