Index: src/wcs1_0/src/main/java/org/geoserver/wcs/response/Wcs10DescribeCoverageTransformer.java
===================================================================
--- src/wcs1_0/src/main/java/org/geoserver/wcs/response/Wcs10DescribeCoverageTransformer.java	(revision 16592)
+++ src/wcs1_0/src/main/java/org/geoserver/wcs/response/Wcs10DescribeCoverageTransformer.java	(working copy)
@@ -509,22 +509,23 @@
             if (metadataNames != null && metadataNames.length > 0) {
                 // TIME DIMENSION
                 elevationMetadata = reader.getMetadataValue("ELEVATION_DOMAIN"); 
-                
-                start("wcs:AxisDescription");
-                element("wcs:name", "ELEVATION");
-                element("wcs:label", "ELEVATION");
-                start("wcs:values");
-                
-                final String [] values=elevationMetadata.split(",");
-                for(String s:values){
-                	element("wcs:singleValue", s);
+                if(elevationMetadata!=null) {
+	                start("wcs:AxisDescription");
+	                element("wcs:name", "ELEVATION");
+	                element("wcs:label", "ELEVATION");
+	                start("wcs:values");
+	                
+	                final String [] values=elevationMetadata.split(",");
+	                for(String s:values){
+	                	element("wcs:singleValue", s);
+	                }
+	                element("wcs:default", values[0]);
+	               
+	               
+	                end("wcs:values");
+	
+	                end("wcs:AxisDescription");
                 }
-                element("wcs:default", values[0]);
-               
-               
-                end("wcs:values");
-
-                end("wcs:AxisDescription");
                 
             }
                         
