Index: src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java
===================================================================
--- src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java  (r▒vision 16353)
+++ src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java  (copie de travail)
@@ -350,7 +350,13 @@
                 continue;

             //build the schema for the types in the single namespace (and don't clean them, they are not dynamic)
-            XSDSchema schema = buildSchemaInternal(new FeatureTypeInfo[] { meta }, null);
+            XSDSchema schema = null;
+           try {
+              schema = buildSchemaInternal(new FeatureTypeInfo[] { meta }, null);
+            }
+            catch (org.springframework.security.AccessDeniedException e) {
+              continue;
+            }

             //declare the namespace
             String prefix = meta.getNamespace().getPrefix();

